en/services/camera.md

```diff
@@ -1,42 +1,74 @@
+The camera protocol is used to configure camera payloads and request their status. 
+It supports photo capture, and video capture and streaming.
+It also includes messages to query and configure the onboard camera storage.
 
 > **Tip** The [Dronecode Camera Manager](https://camera-manager.dronecode.org/en/) provides an implementation of this protocol.
 
 ## Camera Identification
 
-The first step is to determine if a camera exists. Camera components are supposed to send heartbeats just like any other component. There are pre-defined component IDs for cameras - see [MAV_COMP_ID_CAMERA](../messages/common.md#MAV_COMP_ID_CAMERA). If a camera component exists, once a heartbeat is received a [MAV_CMD_REQUEST_CAMERA_INFORMATION](../messages/common.md#MAV_CMD_REQUEST_CAMERA_INFORMATION) message is sent from the GCS. The camera component will then reply with a [CAMERA_INFORMATION](../messages/common.md#CAMERA_INFORMATION) message.
+The first step is to determine what cameras are available/exist.
 
-This response contains the bare minimum information about the camera and what it can or cannot do. By itself, it is sufficient for default image and/or video capture. However, if a camera provides finer control over its settings, this message will also include an URI to a [Camera Definition File](../services/camera_def.md). If this URI exists, the GCS will request it (using a standard HTTP GET request), parse it and prepare the UI for the user to control the camera settings. The definition file can be *hosted* anywhere. If the camera component provides an HTTP interface, the definition file can be hosted on the camera itself. Otherwise, it can be hosted by any regular, reachable server. The `CAMERA_INFORMATION` message should provide a version for the definition file (`cam_definition_version`), allowing the GCS to cache it. Once downloaded, it would only be requested again if the version number changes.
+Camera components must send [heartbeats](../services/heartbeat.md) (just like any other component) and use one of the pre-defined camera component IDs: [MAV_COMP_ID_CAMERA](../messages/common.md#MAV_COMP_ID_CAMERA) to [MAV_COMP_ID_CAMERA6](../messages/common.md#MAV_COMP_ID_CAMERA6).
+The first time a heartbeat is received from a camera component the GCS will send the camera a [MAV_CMD_REQUEST_CAMERA_INFORMATION](../messages/common.md#MAV_CMD_REQUEST_CAMERA_INFORMATION) message. 
+The camera component will then reply with a [CAMERA_INFORMATION](../messages/common.md#CAMERA_INFORMATION) message.
+If no response is sent for a `MAV_CMD_REQUEST_CAMERA_INFORMATION` message, it is assumed camera support is not available and no support for it will be provided by the GCS. 
 
-> **Note** If no response is sent for a `MAV_CMD_REQUEST_CAMERA_INFORMATION` message, it is assumed camera support is not available and no support for it will be provided by the GCS.
+{% mermaid %}
```

Comment:
YOu can see the rendered version of this [here](https://hamishwillee.gitbooks.io/ham_mavdevguide/content/v/cameraupdates/en/services/camera.html#camera-identification).

Docs seem to indicate:
1. CAMERA_INFORMATION not received back means that camera API not supported. 
1. MAV_CMD_REQUEST_CAMERA_INFORMATION is not resent - if message lost, then it's gone and we just assume camera API not supported
1. This is only sent for first heartbeat from camera component.
Is that correct? What is the timeout (or is there one). 

As an aside, I like these diagrams. They give a good opportunity to question whether flow is correctly documented.

Reply:
I have to go back and double check. The idea is that if the message (MAV_CMD_REQUEST_CAMERA_INFORMATION) is ACK'ed, QGC will keep asking until it receives the `CAMERA_INFORMATION` message. This is how the video streaming protocol works (it's fresh in my head). 

> As an aside, I like these diagrams. They give a good opportunity to question whether flow is correctly documented.

Yes, I thought about it while I was adding the video stuff but I didn't know how to add one.

In summary:

GCS sends MAV_CMD_REQUEST_CAMERA_INFORMATION after heartbeat.
Standard protocol will retry the command 3 times if it gets no response (ACK or NAK).
If ACK is received, it will wait for the `CAMERA_INFORMATION` message. If it doesn't get it, it will keep trying (also 3 times). Note that if the `count` field is more than one, the GCS will wait for that count of messages, one for each camera.

The same applies to video info.

Reply:
Regarding diagrams, if you don't want to set up the build environment and just rely on a text editor, the easiest way to render them is to use https://mermaidjs.github.io/mermaid-live-editor/. Basically you open that URL, copy and existing diagram like the one below, omitting the open and close "mermaid tags" and past into the block labeled code on the side. Edit the sequence info (pretty easy) and then copy past back into your doc with mermaid tags. 

```
{% mermaid %}
sequenceDiagram;
    participant GCS
    participant Camera
    Camera->>GCS: HEARTBEAT [cmp id: MAV_COMP_ID_CAMERA] (first) 
    GCS->>Camera: MAV_CMD_REQUEST_CAMERA_INFORMATION
    GCS->>GCS: Start timeout
    Camera->>GCS: CAMERA_INFORMATION
{% endmermaid %}
```

Reply:
> GCS sends MAV_CMD_REQUEST_CAMERA_INFORMATION after heartbeat. 

Yes, but which heartbeat - just the first one? Or to put it another way I assume that this whole process only happens on first connection to a camera component. After we've got CAMERA_INFORMATION don't try get  it again when the next HEARTBEAT arrives?

Reply:
@dogmaphobic OK, I have re-written this section to capture your point about resends and acks. This should be good to review.